### PR TITLE
Display an error message when the service is unreachable

### PR DESCRIPTION
**Pull request by khuddite**
Fixes #8487 

1. Summary
We block the rest of the UI if we fail to fetch client config from the back-end for whatever reason. Therefore, a white screen is displayed without any feedback whenever that happens.
`ClientConfigProvider` is the component that blocks UI when the client config has not been loaded.

2. Solution
It ended up being more changes than I anticipated but those changes were necessary. I removed `ClientConfigProvider` to unblock the rest of the UI even if the back-end is unreachable. Used `enqueueSnackBar` to display an error message when the client config failed to load. Used `isClientConfigLoaded` in a couple of places to ensure no further requests are made if client config fetch fails.

3. Recording

https://github.com/user-attachments/assets/aa719ca5-56b4-46da-a31a-8a8d8f069a29

Honestly, I am unsure if it's the best way to tackle the issue, but I'd like to receive feedback on this.

### CODE REVIEW

File: packages/twenty-front/src/testing/decorators/PageDecorator.tsx [hunk 1]
```diff
@@ -70,26 +69,24 @@ const Providers = () => {
...
-                        </IconsProvider>
-                      </SnackBarProviderScope>
-                    </HelmetProvider>
-                  </FullHeightStorybookLayout>
-                </ObjectMetadataItemsProvider>
-              </ApolloMetadataClientMockedProvider>
-            </UserProvider>
-          </ClientConfigProvider>
+          <UserProviderEffect />
+          <UserProvider>
```
[the remark this answers is not in the document]
Comment (khuddite) [reply packages/twenty-front/src/testing/decorators/PageDecorator.tsx:73]: I don't think it's causing any issues here

**Comment by FelixMalfait** [comment]
Thanks a lot!!!
I'm not sure this is the best solution too. @lucasbordeau what do you think? Should we keep this approach?
One thing that I like is that I feel it could improve performance positively by making this query less render-blocking and starting to display more components sooner.
But on the other hand having the `ClientConfigProviderEffect` so far down the component stack feels wrong as it's almost meant to be "environment variable for the frontend" so something quite generic that's accessible across the code. 
Also the tests broke, not sure if it's easy to fix them @khuddite? Maybe let's wait for confirmation from @lucasbordeau to make sure you don't waste time

**Comment by khuddite** [comment]
> Thanks a lot!!! I'm not sure this is the best solution too. @lucasbordeau what do you think? Should we keep this approach? One thing that I like is that I feel it could improve performance positively by making this query less render-blocking and starting to display more components sooner. But on the other hand having the `ClientConfigProviderEffect` so far down the component stack feels wrong as it's almost meant to be "environment variable for the frontend" so something quite generic that's accessible across the code. Also the tests broke, not sure if it's easy to fix them @khuddite? Maybe let's wait for confirmation from @lucasbordeau to make sure you don't waste time

Thanks for taking a look at this. I have had the same experience while working on this project. When this happened for the first time to me, I was a bit confused and couldn't quite figure out how to resolve it other than running the `npx nx start` command again, even when it's just a matter of order in which services run and hard refresh after few secs would resolve the issue. 
I found the updated behavior a much better experience. I don't think these changes would have any other 
negative implications, such as decreased performance, other than elevated user experience.

**Comment by FelixMalfait** [comment]
@khuddite we discussed it internally and we think it's better not to change the structure like you did but simply reuse `GenericErrorFallback` and display this if we're not able to load the client config. Your change here can have unintended downstream consequence, clientConfig is something we want to be able to access through pretty much every component.

Do you want us to take care of it or do you want to give it another try?

Thanks!

**Comment by khuddite** [comment]
> @khuddite we discussed it internally and we think it's better not to change the structure like you did but simply reuse `GenericErrorFallback` and display this if we're not able to load the client config. Your change here can have unintended downstream consequence, clientConfig is something we want to be able to access through pretty much every component.
> 
> Do you want us to take care of it or do you want to give it another try?
> 
> Thanks!

Thanks for providing the direction. Yeah, I've also noticed that something is not quite correct when I saw many storybook tests have failed for different reasons in consequence. And fixing them didn't seem straightforward.

I will give it another try.

**Comment by FelixMalfait** [comment]
Great thanks a lot!

**Comment by charlesBochet** [comment]
@khuddite Thanks for the PR! We will review it shortly, in the meantime, could you fix the conflicts so we can test it functionally (sorry, the project is still evolving quite fast!)

**Comment by khuddite** [comment]
@charlesBochet I am so sorry, I should've converted it to a draft. This PR is not ready for review just yet. I am still in the middle of making changes and following the direction provided by @FelixMalfait

**Comment by FelixMalfait** [comment]
@khuddite might be easier if you start from scratch again instead of solving the many conflict? Don't mind if I close this one and you re-open it with the simplied approach we discussed? Thanks a lot!

**Comment by khuddite** [comment]
> @khuddite might be easier if you start from scratch again instead of solving the many conflict? Don't mind if I close this one and you re-open it with the simplied approach we discussed? Thanks a lot!

That makes sense, sure go ahead!

**Pull request closed without merging**